kia-lib/src/main/kotlin/com/faforever/ice/telemetry/TelemetryClient.kt

```diff
@@ -51,35 +61,66 @@ class TelemetryClient(
     init {
         logger.info {
             "Open the telemetry ui via ${
-                serverBaseUrl.replaceFirst(
-                    "ws",
-                    "http",
-                )
+                serverBaseUrl.replaceFirst("ws", "http")
             }/app.html?gameId=${iceOptions.gameId}&playerId=${iceOptions.userId}"
         }
 
         val uri: URI = URI.create("$serverBaseUrl/adapter/v1/game/${iceOptions.gameId}/player/${iceOptions.userId}")
         websocketClient = TelemetryWebsocketClient(uri)
 
-        connectingFuture = connectAsync()
+        sendingLoopThread = Thread(this::sendingLoop)
+        sendingLoopThread.uncaughtExceptionHandler = Thread.UncaughtExceptionHandler { _: Thread?, e: Throwable? ->
+            logger.error { "Thread sendingLoop crashed unexpectedly: $e" }
```

Comment:
_ and e are not nullable.
Also
logger.error(e) {...} (and remove $e from the text)